ballerina/graphql_client.bal

```diff
@@ -0,0 +1,183 @@
+                       if (responseMap.hasKey("extensions")) {
+                          responseData = check responseData.mergeJson({ "__extensions" : responseMap.get("extensions") });
+                       }
+                       DataResponse response = check responseData.cloneWithType(targetType);
+                       return response;
+                    } else if targetType is typedesc<record{}> {
+                        record{} response = check httpResponse.cloneWithType(targetType);
+                        return response;
+                    } else if targetType is typedesc<json> {
+                        json response = check httpResponse.cloneWithType(targetType);
+                        return response;
+                    } else {
+                        return error ClientError("GraphQL Client Error, Invalid binding type.");
+                    }
+                }
+            }
+        } on fail var e {
+            return error ClientError("GraphQL Client Error", e);
+        }
+    }
+
+    # Executes a GraphQL query and data binds the GraphQL response to a record with data, extensions and errors 
+    # which is a subtype of GraphQLGenericResponse. The errors of the GraphQL response are directed in the Ballerina 
+    # target type data binding path.
+    #
+    # + query - The GraphQL query. For example `query countryByCode($code:ID!) {country(code:$code) {name}}`.
+    # + variables - The GraphQL variables. For example `{"code": "<variable_value>"}`.
+    # + headers - The GraphQL API headers to execute each query 
+    # + targetType - The payload (`GraphQLGenericResponse`), which is expected to be returned after data binding. For example 
+    #               `type CountryByCodeResponse record {| map<json?> __extensions?; record {| record{|string name;|}? country; |} data; GraphQLError[] errors?; |};`
+    # + return - The payload (if the `targetType` is configured) or a `graphql:GError` if failed to execute the query
+    remote isolated function execute(string query, map<anydata>? variables = (), map<string|string[]>? headers = (), 
+                                     typedesc<GraphQLGenericResponse|GraphQLDataResponse|record{}|json> targetType = <>) 
+                                     returns targetType|GError = @java:Method {
+        'class: "io.ballerina.stdlib.graphql.client.QueryExecutor",
+        name: "execute"
+    } external;
+
+    private isolated function processExecute(typedesc<GraphQLGenericResponse|GraphQLDataResponse|record{}|json> targetType, 
+                                             string query, map<anydata>? variables, map<string|string[]>? headers) 
+                                             returns GraphQLGenericResponse|GraphQLDataResponse|record{}|json|GError {
+        http:Request request = new;
+        json graphqlPayload = getGraphqlPayload(query, variables);
+        request.setPayload(graphqlPayload);
+
```

Comment:
Extra empty line?

Reply:
The extra empty line is removed. The changes are amended. Thank you.